corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/session_middleware.js

```diff
@@ -0,0 +1,37 @@
+/*global FormplayerFrontend */
+
+FormplayerFrontend.module("SessionNavigate", function (SessionNavigate, FormplayerFrontend, Backbone, Marionette) {
+    SessionNavigate.Middleware = {
+        apply: function(api) {
+            var wrappedApi = {};
+            _.each(api, function(value, key) {
+                wrappedApi[key] = function(arguments) {
+                    _.each(SessionNavigate.Middleware.middlewares, function(fn) {
+                        fn(key, arguments);
+                    });
+                    return value(arguments);
+                };
+            });
+            return wrappedApi;
+        },
+    };
+    var backButtonShowHideMiddleware = function(name) {
+        if (name === 'singleApp') {
+            FormplayerFrontend.trigger('phone:back:hide');
+        } else {
+            FormplayerFrontend.trigger('phone:back:show');
+        }
+    };
+    var logRouteMiddleware = function(name) {
+        console.log('User navigated to ' + name);
```

Comment:
I think you can get around this by using `window.console.log` there's a couple of those in vellum